mixer/pkg/lang/compiler/compiler.go

```diff
@@ -366,6 +374,187 @@ func (g *generator) generateEq(f *ast.Function, depth int) {
 	}
 }
 
+func (g *generator) generateLt(f *ast.Function, depth int) {
+	exprType := g.evalType(f.Args[0])
+	g.generate(f.Args[0], depth+1, nmNone, "")
+
+	var constArg1 interface{}
+	if f.Args[1].Const != nil {
+		constArg1 = f.Args[1].Const.Value
+	} else {
+		g.generate(f.Args[1], depth+1, nmNone, "")
+	}
+
+	switch exprType {
+
+	case il.String:
+		if constArg1 != nil {
+			g.builder.ALTString(constArg1.(string))
+		} else {
+			g.builder.LTString()
+		}
+
+	case il.Integer:
+		if constArg1 != nil {
+			g.builder.ALTInteger(constArg1.(int64))
+		} else {
+			g.builder.LTInteger()
+		}
+
+	case il.Double:
+		if constArg1 != nil {
+			g.builder.ALTDouble(constArg1.(float64))
+		} else {
+			g.builder.LTDouble()
+		}
+
+	case il.Interface:
+		dvt, _ := f.Args[0].EvalType(g.finder, g.functions)
+		switch dvt {
+		case descriptor.TIMESTAMP:
+			g.builder.Call("timestamp_lt")
+		default:
+			g.internalError("equality for type not yet implemented: %v", exprType)
```

Comment:
equality =>less than.

There are some copy-paste error in the error messages overall.